pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -308,3 +316,17 @@ func (r *Reconciler) reconcileDomainClaim(ctx context.Context, dm *v1alpha1.Doma
 	dm.Status.MarkDomainClaimed()
 	return nil
 }
+
+func (r *Reconciler) createDomainClaimIfAllowed(ctx context.Context, dm *v1alpha1.DomainMapping) (*netv1alpha1.ClusterDomainClaim, error) {
+	if !config.FromContext(ctx).Network.AutocreateClusterDomainClaims {
+		dm.Status.MarkDomainClaimNotOwned()
+		return nil, fmt.Errorf("domain mapping: namespace %q does not own cluster domain claim for %q", dm.Namespace, dm.Name)
+	}
+
+	dc, err := r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Create(ctx, resources.MakeDomainClaim(dm), metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ClusterDomainClaim: %w", err)
```

Comment:
nit: one of these error has `"domain mapping:"` prefix while the other doesn't

Reply:
also one says "cluster domain claim" while the other uses "ClusterDomainClaim"

Reply:
fixed, dropped the prefix since it doesn't look like we use them elsewhere for stuff like this